app/validators/blocks/block_validator.py

```diff
@@ -62,3 +68,37 @@ def validate_answer_source_reference(self, identifiers):
                 == self.block["id"]
             ):
                 self.add_error(self.ANSWER_SELF_REFERENCE, referenced_id=identifier)
+
+    def validate_redirect_to_list_add_block_params(self):
+        collector_questions = self.questionnaire_schema.get_all_questions_for_block(
+            self.block
+        )
+
+        for collector_question in collector_questions:
+            for collector_answer in collector_question["answers"]:
+                for option in collector_answer.get("options", []):
+                    action = option.get("action")
+
```

Comment:
The validation after this point should only be done if the action type is `RedirectToListAddBlock`.